Xamarin.Essentials/Browser/Browser.shared.cs

```diff
@@ -35,6 +35,60 @@ internal static Uri EscapeUri(Uri uri)
         }
     }
 
+    public readonly partial struct BrowserLaunchOptions : IEquatable<BrowserLaunchOptions>
```

Comment:
I think this should be a `class` - options/config are classes, results/returns are structs.

Also, rather than having multiple parameters, add a `BrowserLaunchMode` property here and add another constructor that takes just this:

```csharp
public BrowserLaunchOptions() {
    // preferred to match the default values in the OpenAsync
    BrowserLaunchMode = BrowserLaunchMode.SystemPreferred;
}

public BrowserLaunchOptions(BrowserLaunchMode launchMode) {
    BrowserLaunchMode = launchMode;
}

public BrowserLaunchOptions(EssentialsColor? preferredControlColor, EssentialsColor? preferredBackgroundColor, EssentialsColor? preferredTitleColor, bool? shouldShowTitle) {

    // Default to this as no one should be setting title color and then
    // saying: just open the browser
    // If this really is a case, then they can just set the property manually
    BrowserLaunchMode = BrowserLaunchMode.SystemPreferred;

    PreferredControlColor = preferredControlColor;
    PreferredBackgroundColor = preferredBackgroundColor;
    PreferredTitleColor = preferredTitleColor;
    ShouldShowTitle = shouldShowTitle;
}
```

Take inspiration from this style: https://github.com/xamarin/Essentials/blob/b018723477f848b108cb9a67adecb3608d0ab686/Xamarin.Essentials/Map/MapLaunchOptions.shared.cs

Reply:
The main difference here is that MapLaunchOptions contains a reference type, thus there is no harm in having it as a reference type as well. But here, we only store structs and bools which in turn are all value types. As the struct is bigger than the size of a reference on all platforms I would rather have it as an In parameter, which would elide copying. Also, I would strongly suggest to keep the immutability of this struct. There should be no reason to modify a BrowserLaunchOptions instance after it was created (at least from a users point I see no use in making it modifyable).

Reply:
I agree with removing the BrowserLaunchMode as a function argument and having it as a meber in the new options struct. Smart solution, I like it!